include/ensmallen_bits/sgd/decay_policies/drop_decay.hpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file drop_decay.hpp
+ * @author Gaurav Sharma
+ *
+ * Drop based decay policy for Stochastic Gradient Descent. 
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_SGD_DECAY_POLICIES_DROP_DECAY_HPP
+#define ENSMALLEN_SGD_DECAY_POLICIES_DROP_DECAY_HPP
+
+#include <cmath>
+
+namespace ens {
+
+/**
+ * Definition of the DropDecay class.
+ */
+class DropDecay
+{
+ public: 
+  /**
+   * This constructor is called before the first iteration.
+   * @param initialStepSize Step Size at the beginning.
+   * @param dropRate Factor by which stepSize is dropped.
+   * @param epochDrop Number of epochs after which stepSize is dropped.
+   */	
+	DropDecay(const double initialStepSize = 0.01,
+			  const double dropRate = 0.1,
+			  const size_t epochDrop = 1000) :
+	initialStepSize(initialStepSize),
+	dropRate(dropRate),
+	epochDrop(epochDrop),
+	epoch(0),
+	effectiveBatchSize(1)
+	{ /* Nothing to do here. */}
+	
+  /**
+   * This function is called in each iteration after the policy update.
+   * It sets the value of effective batch size.
+   *
+   * @param effectiveBatchSize current effective batch size.
+   */	
+  void setEffectiveBatchSize(const size_t& effBatchSize)
+  {
+	effectiveBatchSize = effBatchSize;
+  }

```

Comment:
I added this new function setEffectiveBatchSize to the decay policy template so to get the effectiveBatchSize and update the epoch according to it. Although initially I didn't make any changes to the decay policy template provided by author @zoq  in no_decay.hpp and everything indeed worked by just doing ++epoch. But I personally think that this approach is better and  more robust.